src/main/java/org/micromanager/acqj/internal/Engine.java

```diff
@@ -773,33 +773,23 @@ private void prepareHardware(final AcquisitionEvent event,
          public void run() {
             try {
                // TODO implement Z sequencing for other devices
-//               if (event.isZSequenced()) {
-//                  core_.startStageSequence(zStage);
-//               } else  {
-               for (String stageDeviceName : event.getStageDeviceNames()) {
-                  // TODO: could reimplement logic here to decide when to try to move
-//                  Double previousZ = lastEvent_ == null ? null :
-//                        lastEvent_.getSequence() == null ? lastEvent_.getZPosition() :
-//                              lastEvent_.getSequence().get(0).getZPosition();
-//                  Double currentZ = event.getSequence() == null ? event.getZPosition() :
-//                        event.getSequence().get(0).getZPosition();
-//                  if (currentZ == null) {
-//                     return;
-//                  }
-//                  boolean change = previousZ == null || !previousZ.equals(currentZ);
-//                  if (!change) {
-//                     return;
-//                  }
 
+               // instead of complicated logic in XY stage to only move if the position
+               // changed, just send the command and trust the device to ignore it if it's the same
+               AcquisitionEvent tmpEvent = event;
+               if (event.getSequence() != null && event.getSequence().size() > 0) {
+                  tmpEvent = event.getSequence().get(0);

```

Comment:
Does this mean no sequences over XY stages, because you're just taking the first position? The core does support XY stage sequences I think (though maybe they're never used)